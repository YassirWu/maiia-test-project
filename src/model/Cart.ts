import Product from "./Product";

type Cart = {
  product: Product,
  quantity: number,
}

export default Cart;
