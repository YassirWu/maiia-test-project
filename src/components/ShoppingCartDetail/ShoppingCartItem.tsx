import React from 'react';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Button from 'react-bootstrap/Button';
import Cart from 'model/Cart';
import { makeStyles } from '@material-ui/styles';

const useStyle = makeStyles({
  root: {
    marginBottom: '20px',
  },
  image: {
    width: '100%',
  },
  button: {
    marginRight: '15px',
    borderRadius: '20px'
  }
})

type ShoppingCartItemProps = {
  cart: Cart;
  add: () => void;
  remove: () => void;
}

const ShoppingCartItem: React.FunctionComponent<ShoppingCartItemProps> = ({ cart, add, remove }) => {
  const classes = useStyle();

  return (
    <div className={classes.root}>
      <Row>
        <Col xs={3}>
          <img src={cart.product.thumbnailUrl} alt={cart.product.title} className={classes.image} />
        </Col>
        <Col xs={9}>
          <h6>{cart.product.title}</h6>
          <p>Quantity: {cart.quantity}</p>
          <div>
            <Button onClick={remove} variant="light" className={classes.button}>-</Button>
            <Button onClick={add} variant="light" className={classes.button}>+</Button>
          </div>
        </Col>
      </Row>
    </div>
  );
};

export default ShoppingCartItem;
