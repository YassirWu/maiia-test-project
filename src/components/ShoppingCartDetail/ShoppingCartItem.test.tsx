import React from 'react';
import { shallow } from 'enzyme';
import ShoppingCartItem from './ShoppingCartItem';
import Cart from 'model/Cart';

describe('<ShoppingCartItem />', () => {
  it('should render ShoppingCartItem component', () => {
    const cart: Cart = {
      product: {
        id: 1,
        albumId: 1,
        title: 'my product',
        url: 'url',
        thumbnailUrl: 'thumbnailUrl',
      },
      quantity: 7,
    };
    
    const wrapper = shallow(<ShoppingCartItem cart={cart} add={jest.fn} remove={jest.fn} />);

    expect(wrapper).toMatchSnapshot();
  });
})


