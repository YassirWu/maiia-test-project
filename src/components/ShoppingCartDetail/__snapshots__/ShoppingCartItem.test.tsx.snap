// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShoppingCartItem /> should render ShoppingCartItem component 1`] = `ShallowWrapper {}`;
